ledger/src/test_helpers/chain_builder.rs

```diff
@@ -65,15 +69,36 @@ pub struct TestChainBuilder<N: Network> {
     last_committed_batch_round: HashMap<usize, u64>,
     /// The start of the test chain.
     genesis_block: Block<N>,
+    /// Preloaded transactions to populate the blocks with.
+    transactions: Vec<Transaction<N>>,
 }
 
 /// Additional options you can pass to the builder when generating a set of blocks.
-#[derive(Clone, Default)]
-pub struct GenerateBlocksOptions {
+#[derive(Clone)]
+pub struct GenerateBlocksOptions<N: Network> {
     /// Do not include votes to the previous leader certificate
     pub skip_votes: bool,
     /// Do not generate certificates for the specific node indices (to simulate a partition).
     pub skip_nodes: Vec<usize>,
+    /// A flag indicating that a number of initial "placeholder blocks" should be baked
+    /// wthout transactions in order to skip to the latest version of consensus.
+    pub skip_to_current_version: bool,
+    /// The number of validators.
+    pub num_validators: usize,
+    /// Preloaded transactions to populate the blocks with.
+    pub transactions: Vec<Transaction<N>>,
+}
+
+impl<N: Network> Default for GenerateBlocksOptions<N> {
+    fn default() -> Self {
+        Self {
+            skip_votes: false,
+            skip_nodes: Default::default(),
+            skip_to_current_version: false,
+            num_validators: 0,
```

Comment:
This value is never used - it gets overwritten with the number of private keys in the `TestChainBuilder`.